content/documentation/integration_maximizing_transaction_size.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Maximizing Transaction Size
+description: Integration Guide | Maximizing Transaction Size | Iron Fish Documentation
+---
+
+Iron Fish transactions, in their simplest form, spend notes to fund payments (plus the miners fee).
+The Iron Fish [block](/use/get-started/glossary#Block) has a predefined size limit, so there is a limit on the size and number of transactions in a block.
+In effect, if you are maximizing transaction size, you are trying to make the transaction size equal to the max
+block size.
+
+Here's an example that creates a transaction with a rough estimate of the number of notes that can fit in a block, then incrementally removes notes until the transaction will fit in a block.
+
+```js
+import { Asset } from '@ironfish/rust-nodejs'
+import {
+  CreateTransactionRequest,
+  CurrencyUtils,
+  IronfishSdk,
+  RawTransactionSerde,
+} from '@ironfish/sdk'
+import { NOTE_ENCRYPTED_SERIALIZED_SIZE_IN_BYTE } from '@ironfish/sdk/src/primitives/noteEncrypted'
+async function main(): Promise<void> {
+  // creating fake outputs for example
+  const outputs: CreateTransactionRequest['outputs'] = []
+  for (let i = 0; i < 100000; i++) {
```

Comment:
I think it'd be a bit clearer to move the output creation to after calculating lengthGuess, so we only need to generate `lengthGuess` outputs instead of 100000